### PR TITLE
[DataGrid] Add docs for Column selector

### PR DESCRIPTION
**Pull request by DanailH**
Part 1 (docs) of 2 that is needed to wrap up #345

### CODE REVIEW

File: docs/src/pages/components/data-grid/columns/columns.md [hunk 1]
```diff
@@ -125,6 +125,12 @@ By default, each column header displays a column menu. The column menu allows ac
 
 {{"demo": "pages/components/data-grid/columns/ColumnMenuGrid.js", "bg": "inline"}}
 
+## Column selector
+
+To enable the column selector you need to add the showToolbar prop to the data grid. The user can choose which columns to be visible using the column selector from the toolbar.
```
Comment (dtassone) [inline docs/src/pages/components/data-grid/columns/columns.md:130]: there is also a link in the column menu

Comment (DanailH) [reply docs/src/pages/components/data-grid/columns/columns.md]: Updated

**Review by dtassone — APPROVED** [review]

File: docs/src/pages/components/data-grid/columns/columns.md [hunk 2]
```diff
@@ -125,6 +125,12 @@ By default, each column header displays a column menu. The column menu allows ac
 
 {{"demo": "pages/components/data-grid/columns/ColumnMenuGrid.js", "bg": "inline"}}
 
+## Column selector
+
+To enable the column selector in the toolbar you need to add the showToolbar prop to the data grid. In addition the column selector can be shown by using the `Show columns` button in the column menu. The user can choose which columns to be visible using the column selector from the toolbar.
```
Comment (oliviertassinari) [inline docs/src/pages/components/data-grid/columns/columns.md:130]:
```suggestion
To enable the column selector in the toolbar you need to add the `showToolbar` prop to the data grid.
In addition, the column selector can be shown by using the "Show columns" menu item in the column menu.
The user can choose which columns to be visible using the column selector from the toolbar.
```

File: packages/grid/_modules_/grid/components/panel/Panel.tsx [hunk 3]
```diff
@@ -57,7 +57,16 @@ export function Panel(props: PanelProps) {
...
 
   return (
-    <Popper placement="bottom-start" open={open} anchorEl={anchorEl}>
+    <Popper
+      placement="bottom-start"
+      open={open}
+      anchorEl={anchorEl}
+      modifiers={{
+        flip: {
+          enabled: false,
```
Comment (oliviertassinari) [inline packages/grid/_modules_/grid/components/panel/Panel.tsx:66]: 👍

Comment (oliviertassinari) [reply packages/grid/_modules_/grid/components/panel/Panel.tsx:66]: Mind that this will need special handling between Material-UI v4 and v5

**Review by oliviertassinari — COMMENTED** [review]
Looks good, it's a bit weird that we display all the items in the toolbar, but I guess #726 will allow solving this.

File: packages/grid/_modules_/grid/components/panel/Panel.tsx (hunk 3 above)
Comment (DanailH) [reply packages/grid/_modules_/grid/components/panel/Panel.tsx:66]: Yup, I remember that you suggested to disable it. I'll check how to handle it for v5

**Comment by DanailH** [comment]
> Looks good, it's a bit weird that we display all the items in the toolbar, but I guess #726 will allow solving this.

Yes I also didn't like it much but after the composable toolbar is done we can make the demo more purposeful.

**Review by oliviertassinari — APPROVED** [review]

**Pull request merged**